streams/src/main/java/org/apache/kafka/streams/state/internals/NamedCache.java

```diff
@@ -358,28 +359,46 @@ private void update(LRUCacheEntry entry) {
         final StreamsMetricsImpl metrics;
         final String groupName;
         final Map<String, String> metricTags;
+        final Map<String, String> allMetricTags;
         final Sensor hitRatioSensor;
 
+        private void maybeAddMetric(Sensor sensor, MetricName name, MeasurableStat stat) {
+            if (!metrics.metrics().containsKey(name)) {
+                sensor.add(name, stat);
+            } else {
+                log.trace("Trying to add metric twice: {}", name);
+            }
+        }
 
         public NamedCacheMetrics(StreamsMetrics metrics) {
             final String scope = "record-cache";
-            final String entityName = name;
             final String opName = "hitRatio";
             final String tagKey = "record-cache-id";
-            final String tagValue = name;
+            final String tagValue = ThreadCache.underlyingStoreNamefromCacheName(name);
             this.groupName = "stream-" + scope + "-metrics";
             this.metrics = (StreamsMetricsImpl) metrics;
-            this.metricTags = new LinkedHashMap<>();
-            this.metricTags.put(tagKey, tagValue);
-
-            hitRatioSensor = this.metrics.registry().sensor(entityName + "-" + opName, Sensor.RecordingLevel.DEBUG);
-
-            hitRatioSensor.add(this.metrics.registry().metricName(entityName + "-" + opName + "-avg", groupName,
-                "The average cache hit ratio of " + entityName, metricTags), new Avg());
-            hitRatioSensor.add(this.metrics.registry().metricName(entityName + "-" + opName + "-min", groupName,
-                "The minimum cache hit ratio of " + entityName, metricTags), new Min());
-            hitRatioSensor.add(this.metrics.registry().metricName(entityName + "-" + opName + "-max", groupName,
-                "The maximum cache hit ratio of " + entityName, metricTags), new Max());
+            this.allMetricTags = ((StreamsMetricsImpl) metrics).tagMap(tagKey, "all",
+                    "task-id", ThreadCache.taskIDfromCacheName(name));
+            this.metricTags = ((StreamsMetricsImpl) metrics).tagMap(tagKey, tagValue,
+                    "task-id", ThreadCache.taskIDfromCacheName(name));
+
+            // add parent
+            Sensor parent = this.metrics.registry().sensor(opName, Sensor.RecordingLevel.DEBUG);
+            maybeAddMetric(parent, this.metrics.registry().metricName(opName + "-avg", groupName,
+                    "The average cache hit ratio of " + name, allMetricTags), new Avg());
```

Comment:
nit: we do not need the `of name` part anymore in the metric description, since it is already under the tags. Just `The average cache hit ratio` is good enough. Ditto elsewhere.